timesketch/api/v1/resources/archive.py

```diff
@@ -424,96 +442,364 @@ def _export_view(self, view, sketch, zip_file):
+           - If the OpenSearch index exists but an error occurs during the close
+             operation (other than not found), the SearchIndex's database status
+             remains unchanged (e.g., 'ready' or 'fail'). This discrepancy allows
+             tools like `tsctl list-sketches --archived-with-open-indexes`
+             to identify potential issues requiring administrative attention.
+
+        This is a non-destructive operation; it does not delete any event data.
 
         Args:
-            sketch: Instance of timesketch.models.sketch.Sketch
+            sketch (Sketch): Instance of timesketch.models.sketch.Sketch
         """
-        if sketch.get_status.status != "archived":
+        # 1. Pre flight checks
+        if sketch.get_status.status == "archived":
             abort(
                 HTTP_STATUS_CODE_BAD_REQUEST,
-                "Unable to unarchive a sketch that wasn't already archived "
-                "(sketch: {:d})".format(sketch.id),
+                f"Sketch {sketch.id} is already archived.",
             )
 
-        sketch.set_status(status="ready")
+        labels_to_prevent_deletion = current_app.config.get(
+            "LABELS_TO_PREVENT_DELETION", []
+        )
 
-        indexes_to_open = []
-        for timeline in sketch.timelines:
-            if timeline.get_status.status != "archived":
+        for label in labels_to_prevent_deletion:
+            if sketch.has_label(label):
+                abort(
+                    HTTP_STATUS_CODE_FORBIDDEN,
+                    f"Sketch {sketch.id} has label '{label}'"
+                    " and cannot be archived.",
+                )
+
+        errors_occurred = False
+        error_details = []
+
+        # Check if any timeline is in a state that prevents archiving.
+        statuses_preventing_archival = ["processing", "fail", "timeout"]
+        for timeline_to_check in sketch.timelines:
+            timeline_status = timeline_to_check.get_status.status
+            if timeline_status in statuses_preventing_archival:
+                base_error_msg = (
+                    f"Cannot archive sketch {sketch.id}. Timeline "
+                    f"'{timeline_to_check.name}' (ID: {timeline_to_check.id}) is in "
+                    f"a non-archivable state: '{timeline_status}'."
+                )
+                suggestion = ""
+                if timeline_status in ["fail", "timeout"]:
+                    suggestion = " Please delete this timeline and try again."
+                elif timeline_status == "processing":
+                    suggestion = (
+                        " Please wait for it to finish processing. If it seems to be "
+                        "stuck, contact your system administrator to resolve the "
+                        "issue."
+                    )
+                error_msg = f"{base_error_msg}{suggestion}"
+                logger.error(error_msg)
```

Comment:
Can we prevent that `timeline_to_check.name` gets posted to the backend log?
But we should include it for the `abort` message so the user knows what to do.